src/partitioner/stream/HashPartitioner.cpp

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2024 JasminGraph Team
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "HashPartitioner.h"
+#include "../../server/JasmineGraphServer.h"
+#include <nlohmann/json.hpp>
+
+Logger hash_partitioner_logger;
+
+int PARTITION_FILE_EDGE_COUNT_THRESHOLD = 1000000;
+
+HashPartitioner::HashPartitioner(int numberOfPartitions, int graphID, std::string masterIp)
+        : numberOfPartitions(numberOfPartitions), graphId(graphID),
+          partitionLocks(numberOfPartitions), vertexCount(0), edgeCount(0),
+          localEdgeArrays(numberOfPartitions), edgeCutsArrays(numberOfPartitions),
+          localEdgeMutexes(numberOfPartitions), edgeAvailableCV(numberOfPartitions),
+          edgeReady(numberOfPartitions, false), edgeCutsMutexes(numberOfPartitions),
+          edgeCutsAvailableCV(numberOfPartitions), edgeCutsReady(numberOfPartitions, false),
+          terminateConsumers(false), masterIp(masterIp), partitionMutexArray(numberOfPartitions) {
+    this->outputFilePath = Utils::getHomeDir() + "/.jasminegraph/tmp/hdfsstore/" + std::to_string(this->graphId);
```

Comment:
We should not persist data under the JasmineGraph Home. Better keep this data in mounted location. See https://github.com/miyurud/jasminegraph/blob/master/docker-compose.yml#L13 for example.